receiver/mongodbatlasreceiver/events.go

```diff
@@ -0,0 +1,335 @@
+	}
+
+	return r
+}
+
+func (er *eventsReceiver) Start(ctx context.Context, host component.Host) error {
+	er.logger.Debug("Starting up events receiver")
+	return er.startPolling(ctx, host)
+}
+
+func (er *eventsReceiver) Shutdown(ctx context.Context) error {
+	er.logger.Debug("Shutting down events receiver")
+	close(er.doneChan)
+	er.wg.Wait()
+	return er.checkpoint(ctx)
+}
+
+func (er *eventsReceiver) startPolling(ctx context.Context, host component.Host) error {
+	storageClient, err := adapter.GetStorageClient(ctx, host, er.storageID, er.id)
+	if err != nil {
+		return fmt.Errorf("failed to set up storage: %w", err)
+	}
+	er.storageClient = storageClient
+	err = er.loadCheckpoint(ctx)
+
+	t := time.NewTicker(er.pollInterval)
+	er.wg.Add(1)
+	go func() {
+		defer er.wg.Done()
+		for {
+			select {
+			case <-t.C:
+				if err = er.pollEvents(ctx); err != nil {
+					er.logger.Error("error while polling for events", zap.Error(err))
+				}
+			case <-er.doneChan:
+				return
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (er *eventsReceiver) pollEvents(ctx context.Context) error {
+	for _, p := range er.cfg.Events.Projects {
+		project, err := er.client.GetProject(ctx, p.Name)
+		if err != nil {
+			er.logger.Error("error retrieving project information for "+p.Name+":", zap.Error(err))
+			return err
+		}
+		er.poll(ctx, project, p)
+	}
+
+	return er.checkpoint(ctx)
+}
+
+func (er *eventsReceiver) poll(ctx context.Context, project *mongodbatlas.Project, p *ProjectConfig) {
+	pollTime := time.Now()
+	for pageN := 1; pageN <= er.maxPages; pageN++ {
+		opts := &internal.GetEventsOptions{
+			PageNum:    pageN,
+			EventTypes: er.cfg.Events.Types,
+			MaxDate:    pollTime,
+		}
+
+		if er.record.LastRecordedEvent != nil {
+			opts.MinDate = *er.record.LastRecordedEvent
+		} else {
+			opts.MinDate = time.Now().Add(-er.pollInterval)
+		}
+
+		projectEvents, hasNext, err := er.client.GetEvents(ctx, project.ID, opts)
+		if err != nil {
+			er.logger.Error("unable to get events for project", zap.Error(err), zap.String("project", p.Name))
+			break
+		}
+
+		now := pcommon.NewTimestampFromTime(pollTime)
+		logs := er.transformEvents(now, projectEvents, project)
+
+		if logs.LogRecordCount() > 0 {
+			if err = er.consumer.ConsumeLogs(ctx, logs); err != nil {
+				er.logger.Error("error consuming events", zap.Error(err))
+				break
+			}
+		}
+
+		if !hasNext {
+			break
+		}
+	}
+
+}
+
+func (er *eventsReceiver) transformEvents(now pcommon.Timestamp, events []*mongodbatlas.Event, p *mongodbatlas.Project) plog.Logs {
+	logs := plog.NewLogs()
+	for _, event := range events {
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		ra := resourceLogs.Resource().Attributes()
+		ra.PutStr("mongodbatlas.project.name", p.Name)
+		ra.PutStr("mongodbatlas.org.id", p.OrgID)
+		ra.PutStr("mongodbatlas.group.id", event.GroupID)
+
+		if event.ReplicaSetName != "" {
+			ra.PutStr("mongodbatlas.replica_set.name", event.ReplicaSetName)
+		}
```

Comment:
Should we separate events by group id and replicate set name, so we can put multiple logs on a resource, rather than emitting a resource for every log?

Reply:
Ahh I think that's a good idea 👍